slow_query_exporter.rb

```diff
@@ -0,0 +1,243 @@
+    return "percona-toolkit" if q =~ %r{/\*\w+\.\w+:[0-9]/[0-9]\*/}
+    return q if q =~ /\Aadministrator command: /i
+    return $1.downcase if q =~ /\A\s*(call\s+\S+)\(/i
+
+    # Truncate multi-value INSERT statements
+    q = $1 if q =~ /\A((?:INSERT|REPLACE)(?: IGNORE)?\s+INTO.+?VALUES\s*\(.*?\))\s*,\s*\(/im
+
+    q.gsub!(%r{/\*[^!].*?\*/}m, "")                     # Strip multi-line comments
+    q.gsub!(%r{(?:--|#)[^'"\r\n]*(?=[\r\n]|\Z)}m, "")   # Strip single-line comments
+    return "use ?" if q =~ /\Ause \S+\Z/i
+
+    # Replace quoted strings with "?"
+    q.gsub!(/\\["']/, "")
+    q.gsub!(/".*?"/, "?")
+    q.gsub!(/'.*?'/, "?")
+
+    q.downcase!
+    q.gsub!(/\bfalse\b|\btrue\b/, "?")      # Replace boolean values with "?"
+    q.gsub!(/[0-9+-][0-9a-f.xb+-]*/, "?")   # Replace numbers with "?"
+    q.gsub!(/[xb.+-]\?/, "?")               # Strip number prefixes
+
+    q.gsub!(/\bnull\b/, "?")   # Replace NULLs with "?"
+
+    # Collapse IN and VALUES lists
+    q.gsub!(/\b(in|values?)(?:[\s,]*\([\s?,]*\))+/, "\\1(?+)")
+    q.gsub!(/\b(select\s.*?)(?:(\sunion(?:\sall)?)\s\1)+/, "\\1 /*repeat$2*/")   # Collapse UNIONs
+    q.sub!(/\blimit \?(?:, ?\?| offset \?)?/, "limit ?")   # LIMITs and OFFSETs
+    q.gsub(/\border by (.+?)\s+asc/, "order by \\1")       # Remove extraneous ASCs from ORDER BYs
+  end
+
+  # Remove leading/trailing whitespace and compress all other runs of whitespace into a single space.
+  def compress(str)
+    str.strip.tr_s(" \n\t\r\f", " ")
+  end
+end
+
+
+$graylog_host = "localhost"
+$graylog_port = 12201
+$verbose = false
+
+HELP_TEXT = "Usage: slow_query_exporter.rb [-v] [-h host] [-p port] slow_query.log
+Options:
+    -h, --host      The Graylog host
+    -p, --port      The Graylog port
+    -v, --verbose   Print entries to stdout as they're parsed
+    -?, --help      Display this help text
+"
+
+OptionParser.new do |opts|
+  opts.banner = HELP_TEXT
+
+  opts.on("-h", "--host") { |host| $graylog_host = host }
+  opts.on("-p", "--port") { |port| $graylog_port = port }
+  opts.on("-v", "--verbose") { $verbose = true }
+  opts.on("-?", "--help") do
+    $stderr.puts(HELP_TEXT)
+    exit
+  end
+end.parse!
+
+Syslog.open("slow_query_exporter", Syslog::LOG_PID | Syslog::LOG_PERROR, Syslog::LOG_DAEMON)
+
+if ARGV.empty?
+  $stderr.puts(HELP_TEXT)
+  exit 1
+end
+
+# We touch the logfile to make sure it exists before we start. Otherwise, "tail -F" will die.
+logfile = ARGV[0]
+File.open(logfile, "a") {}
+
+gelf = GELF::Notifier.new($graylog_host, $graylog_port, "WAN")
+parser = QueryParser.new
+
+begin
+  tail = IO.popen(["tail", "-F", "-n", "+0", logfile])
+  tail.each_line do |line|
+    parser.parse_line(line)
+    query = parser.pop_finished_query
+    if query
+      puts query.gelf_attributes.inspect, "\n" if $verbose
+      gelf.notify!(query.gelf_attributes)
+    end
+  end
+
+ensure
+  Process.kill("INT", tail.pid) if tail
+end
```

Comment:
Will this program be run as a daemon as the comment suggests? If so, I fail to see where the daemonization takes place. Is this change forthcoming? 

I'm also wondering whether the program should be structured as a gem. A gem could provide two benefits. Firstly, infrastructure can easily manage and deploy gems.  Secondly, your code would be structured using the conventional gem layout. The executable would be only responsible for option handling, daemonizing, and kicking off exporter.